context/app/static/js/components/workspaces/CreateWorkspaceDialog/CreateWorkspaceDialog.jsx

```diff
@@ -1,28 +1,42 @@
 import React from 'react';
 import Button from '@mui/material/Button';
+import Box from '@mui/material/Box';
 
 import CreateWorkspaceInput from 'js/components/workspaces/CreateWorkspaceInput';
 import DialogModal from 'js/shared-styles/DialogModal';
+import { useStore } from 'js/shared-styles/tables/SelectableTableProvider/store';
 import { useCreateWorkspace } from './hooks';
 
 function CreateWorkspaceDialog({ handleCreateWorkspace, buttonComponent: ButtonComponent, defaultName, ...rest }) {
   const { dialogIsOpen, setDialogIsOpen, handleSubmit, handleClose, control, errors, onSubmit } = useCreateWorkspace({
     handleCreateWorkspace,
     defaultName,
   });
+  const { selectedRows } = useStore();
+  const errorMessage =
+    selectedRows.size > 10
+      ? `You have selected ${selectedRows.size} datasets. Workspaces currently only supports up to 10 datasets. Please unselect datasets.`
+      : null;
+
```

Comment:
We don't expect there to be a `SelectableTableProvider` parent to every `CreateWorkspaceDialog` you need to access the store in `MetadataMenu` and pass the requisite information as props.